tests/compile-fail/invalid_upcast_comparisons.rs

```diff
@@ -0,0 +1,19 @@
+#![feature(plugin)]
+#![plugin(clippy)]
+
+#![deny(invalid_upcast_comparisons)]
+#![allow(unused, eq_op, no_effect)]
+fn main() {
+    let zero: u32 = 0;
+    let u8_max: u8 = 255;
+
+    (u8_max as u32) > 300; //~ERROR Because of the numeric bounds prior to casting, this expression is always false.
```

Comment:
If you feel like it, you could add a "note: `u8_max` has type `u8`" in case the user still doesn’t realize why the comparison would be always true or false.